### ISSUE TITLE
Supporting unattended install on Linux

### ISSUE THREAD

**Issue by urisimchoni** [Azure/azure-cli#3946]
### Description
**Outline the issue here:**
Supporting unattended install on Linux
---

### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Answer here: Linux method - curl -L https://aka.ms/InstallAzureCli | bash

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Answer here: 2.0
**OS Version:** What OS and version are you using?  
Answer here: Fedora 25

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Answer here: bash

Installation works great, but I couldn't figure out how to run an unattended install (like when provisioning a VM) - the questions in the install script halt execution. Would be nice to be able to accept the defaults, or even to define the answers via some method (other than having to type them via the console).

Thanks,
Uri

**Comment by BorisB2015** [comment]
I run into this issue while searching for the very same answer. Hence posting my workaround which may help you/others.

In short, I replaced a couple of lines in the downloaded script to install CLI to read standard input (by default it connects to tty) and then used "here string" to provide input to accept defaults. This seem to have worked as expected. 

[Reference to install Azure CLI 2.0 and its dependencies](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli#a-namelinuxinstall-on-linux-without-apt-get)

[Here strings reference](https://en.wikipedia.org/wiki/Here_document#Here_strings)

The code to install

```` bash
yum check-update; sudo yum install -y gcc python libffi-devel python-devel openssl-devel
curl -L https://aka.ms/InstallAzureCli -o cli-install.sh
chmod +x ./cli-install.sh
sed -i -e "s/^_TTY/#&/;s/< $_TTY/#&/" ./cli-install.sh
./cli-install.sh <<< "



"
````
I used this in Custom Script Extension (which runs as root during VM provision) for OpenShift Origin dependency install.

[Example here](https://github.com/BorisB2015/openshift-origin/commit/3511ac902e77b35d3d2fe6e33fa68a4cd90bb6d4#diff-5122bd37142550977e7f76c928502618). 

Hope that helps!
